### PR TITLE
Fixes/cleans up issues with opening emotes file

### PR DESCRIPTION
**Pull request by mracine**

### CODE REVIEW

File: emote.py [hunk 1]
```diff
@@ -14,11 +14,15 @@ def init(self):
...
-					open(self.path, 'w+')
+					with open(self.path, 'w+') as f:
+						json.dump(self.emotes, f)
 
 				else:
-					self.emotes = json.loads(open(self.path, 'r').read())
+					with open(self.path, 'r') as f:
+						self.emotes = json.loads(f.read())
 
+			except (JSONDecodeError):
```
Comment (mracine) [inline emote.py]: This should be fine, I wasn't getting any errors for this. Just curious that in GitHub and in vim the exception isn't 'recognized' (?) like the IOError and OSError.

Comment (sjrct) [reply emote.py]: Did you trigger that path? Names python `except` statements are not resolved until an error within the try block is actually raised. It looks to me like you are missing a `json.` in front of `JSONDecodeError`. And although the doc says it exists, that exception does not seem to exist in my python3.4 installation... strange.

Comment (richteer) [reply emote.py]: `json.JSONDecodeError` seems to exist for me. That is strange.

Comment (mracine) [reply emote.py]: Just looked it up, apparently in python3.5, json.loads throws a JSONDecodeError if the object is not valid JSON. In python3.4, json.loads throws a ValueError. JSONDecodeError is a subclass of ValueError
https://docs.python.org/3/library/json.html
https://docs.python.org/3/library/json.html#json.JSONDecodeError

Comment (sjrct) [reply emote.py]: Ah yes, that little italic _New in version 3.5_ slipped my gaze.

**Review by sjrct — APPROVED** [review]

File: emote.py (hunk 1 above)
Comment (richteer) [reply emote.py]: Well, that's exciting.

**Pull request merged**